### PR TITLE
Exit on non-existent DB in sparse_merkle_db_editor

### PR DESCRIPTION
**Pull request by dartdart26**
If any of the passed RocksDB DBs to sparse_merkle_db_editor are
non-existent or empty, exit with an error instead of reporting empty
values. Prior to this change, the editor was first creating an empty DB
and then all commands were working by reporting default/empty values.

Since RocksDB doesn't offer a consistent method for checking if a DB
exists at a given path, first check if the passed directory exists. If
it doesn't, report an error. If it does, check if there are any keys in
the DB - if there are none, report an error. Above behavior can be
simplified when the following RocksDB issue is resolved:
https://github.com/facebook/rocksdb/issues/5029

Add tests for non-existent and empty databases for commands accepting
both one and two DB paths.

### CODE REVIEW

**Review by f-squirrel — APPROVED** [review]
LGTM!

File: kvbc/tools/sparse_merkle_db/include/sparse_merkle_db_editor.hpp [hunk 1]
```diff
@@ -37,6 +37,16 @@
 #include <variant>
 #include <vector>
 
+#if __has_include(<filesystem>)
```
Comment (tdimitrov) [inline kvbc/tools/sparse_merkle_db/include/sparse_merkle_db_editor.hpp:40]: Portability to the limit ;)

**Review by tdimitrov — APPROVED** [review]

**Review by andrewjstone — APPROVED** [review]
LGTM. Thanks for the fix @dartdart26
